src/sphinx/reference/adapters.rst

```diff
@@ -54,18 +54,221 @@ For persisting the stream processing progress, an application should store the l
 .. hint::
    The full example source code is in `SparkStreamAdapterExample.scala`_
 
+.. _vertx-adapter:
+
+Vert.x adapter
+--------------
+
+The Eventuate Vert.x adapter allows applications using a `Vert.x`_ instance to interact with event logs. Events can both be consumed from and produced to an event log by a Vert.x instance.
+
+Event exchange is performed over the Vert.x `event bus`_. Events delivered to a Vert.x instance are either published to all subscribers or sent to a single subscriber on the event bus. Events received from a Vert.x instance are persisted to an event log by consuming events from a particular endpoint on the event bus.
+
+Event Producers
+~~~~~~~~~~~~~~~
+
+The Vert.x adapter exchanges events with a Vert.x instance by using so called *event producers*. An event producer consumes events from a given source and produces the same events to a specified destination. Both sources and destinations can either be an event bus endpoint or an even log.
+
+The Vert.x adapter supports two kinds of event producers:
+
+- **Vert.x event producers** consume events from an event log and publish or send the events to a configurable event bus endpoint.
+- **Log event producers** consume events from a given event bus endpoint and persist the events in an event log.
+
+An event producer establishes a unidirectional connection between exactly one event log and one or multiple event bus endpoints. Event producers are instantiated by using the ``EventProducer`` factory methods. The configuration of a producer consists of:
+
+- an event source,
+- an event destination and
+- a unique id.
+
+*Vert.x producers* are created by using the ``EventProducer.fromLog`` method. Applications define the source log, the delivery method and an arbitrary amount of event bus endpoints, the events will be delivered to.
+Vert.x producers support routing of events based on the content of the event.
+
+.. tabbed-code::
+   .. includecode:: ../../../eventuate-example-vertx/src/main/scala/com/rbmhtechnology/docs/vertx/Documentation.scala
+      :snippet: vertx-event-producer
+   .. includecode:: ../../../eventuate-example-vertx/src/main/java/com/rbmhtechnology/docs/vertx/japi/Documentation.java
+      :snippet: vertx-event-producer
+
+*Log producers* are created by using the ``EventProducer.fromEndpoints`` method. Multiple event bus endpoints can be defined, which are used to consume events from the event bus and persist the same events to the given event log. Events can be filtered by defining a filter function.
```

Comment:
Either include an example of a filter function or mention that it is not shown in the example